### PR TITLE
New functions and tweaks

### PR DESCRIPTION
**Pull request by MartinSGill**
- Added Get-Theme function to list available themes
- Added Show-ThemeSymbols function to list defined symbols
- Allow themes to be re-loaded by Set-Theme
- Made Show-ThemeColors implementation more generic
- ArgumentCompleter now re-uses Get-Theme function

### CODE REVIEW

File: oh-my-posh.psd1 [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 # Version number of this module.
 ModuleVersion = '%version%'
+#ModuleVersion = '0.0.0'
```
Comment (JanDeDobbeleer) [inline oh-my-posh.psd1]: This needs to be reset or it won't be build :-)

File: oh-my-posh.psm1 [hunk 2]
```diff
@@ -67,21 +67,35 @@ function global:Write-WithPrompt() {
...
-    Write-ColorPreview -text 'PromptBackgroundColor            ' -color $sl.Colors.PromptBackgroundColor
-    Write-ColorPreview -text 'PromptSymbolColor                ' -color $sl.Colors.PromptSymbolColor
-    Write-ColorPreview -text 'PromptHighlightColor             ' -color $sl.Colors.PromptHighlightColor
-    Write-ColorPreview -text 'SessionInfoBackgroundColor       ' -color $sl.Colors.SessionInfoBackgroundColor
-    Write-ColorPreview -text 'SessionInfoForegroundColor       ' -color $sl.Colors.SessionInfoForegroundColor
-    Write-ColorPreview -text 'CommandFailedIconForegroundColor ' -color $sl.Colors.CommandFailedIconForegroundColor
-    Write-ColorPreview -text 'AdminIconForegroundColor         ' -color $sl.Colors.AdminIconForegroundColor
-    Write-ColorPreview -text 'WithBackgroundColor              ' -color $sl.Colors.WithBackgroundColor
-    Write-ColorPreview -text 'WithForegroundColor              ' -color $sl.Colors.WithForegroundColor
+    $sl.Colors.Keys | Sort-Object | ForEach-Object { Write-ColorPreview -text ("{0,-35}" -f $_ ) -color $sl.Colors[$_] }
```
Comment (JanDeDobbeleer) [inline oh-my-posh.psm1:80]: Makes a lot more sense, I was mainly lazy :-)

File: oh-my-posh.psm1 [hunk 3]
```diff
@@ -120,11 +134,11 @@ function Set-Theme {
...
     else {
         Write-Host ''
-        Write-Host "Theme $name not found. Available themes are:"
-        Show-Themes
+        Write-Warning "Theme $name not found. Available themes are:"
+        Get-Theme
     }
 
-    Set-Prompt
+    Set-Prompt;
```
Comment (JanDeDobbeleer) [inline oh-my-posh.psm1]: Any specific reason to add `;` here?

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

File: oh-my-posh.psm1 (hunk 3 above)
Comment (MartinSGill) [reply oh-my-posh.psm1]: Left over from some debugging code. I suspect.

File: oh-my-posh.psd1 (hunk 1 above)
Comment (MartinSGill) [reply oh-my-posh.psd1]: It's commented out.. but I'll remove it all the same.

Comment (JanDeDobbeleer) [reply oh-my-posh.psd1]: Cool didn't see that.

**Review by JanDeDobbeleer — APPROVED** [review]
Code wise it looks ok, I just need to be able to validate this on a Windows machine (short of one ATM).

**Comment by JanDeDobbeleer** [comment]
Update: I'll have access to my Windows setup on Monday. If everything works as advertised, I'll deploy immediately. And I'll make sure to create a VM setup.

**Comment by MartinSGill** [comment]
I've got a windows setup, does that count? :)

**Comment by JanDeDobbeleer** [comment]
> I've got a windows setup, does that count? :)

Well, as much as I'd love to put all my trust in strangers on the internet, I really want to check everything at least once myself for now :-)

**Pull request merged**